lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -540,15 +580,157 @@ def test_cache_map_bad_dir_path(self, monkeypatch, tmp_path):
                 tmp_path, "dir_name"
             )
             tb.cache_map(tar_dir)
-
-            file_path = "/dir_name/subdir1/f11.txt"
-            expected_msg = f"The directory path {file_path!r} is an absolute path, we expect relative path to the root directory."
             with pytest.raises(BadDirpath) as exc:
                 tb.get_info(Path(file_path))
             assert str(exc.value) == expected_msg
 
-    def test_cache_map_traverse_get_info_cmap(self, monkeypatch, tmp_path):
-        """Test to build the cache map of the unpacked tarball"""
+    @pytest.mark.parametrize(
+        "file_path, expected_msg",
+        [
+            ("dir_name/subdir1/./f11.txt", "f11.txt"),
+            ("dir_name/subdir1//f11.txt", "f11.txt"),
+        ],
+    )
+    def test_cache_map_irregular_path(
+        self, monkeypatch, tmp_path, file_path, expected_msg
+    ):
+        """Test to check irreguraties in file/directory path"""
+        tar = Path("/mock/dir_name.tar.xz")
+        cache = Path("/mock/.cache")
+
+        with monkeypatch.context() as m:
+            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
+            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
+            tar_dir = TestCacheManager.MockController.generate_test_result_tree(
+                tmp_path, "dir_name"
+            )
+            tb.cache_map(tar_dir)
+            tb.get_info(Path(file_path))
+            assert (
+                tb.cachemap["dir_name"]["children"]["subdir1"]["children"]["f11.txt"][
+                    "details"
+                ].name
+                == expected_msg
+            )
+
+    @pytest.mark.parametrize(
+        "file_path, location, name, resolve_path, resolve_type, size, file_type",
+        [
+            ("dir_name", "dir_name", "dir_name", None, None, None, "directory"),
+            (
+                "dir_name/subdir1",
+                "dir_name/subdir1",
+                "subdir1",
+                None,
+                None,
+                None,
+                "directory",
+            ),
+            ("dir_name/f1.json", "dir_name/f1.json", "f1.json", None, None, 0, "file"),
+            (
+                "dir_name/subdir1/subdir13/subdir111/f1112_sym",
+                "dir_name/subdir1/subdir13/subdir111/f1112_sym",
+                "f1112_sym",
+                Path("dir_name/subdir1/f11.txt"),
+                "file",
+                None,
+                "symlink",
+            ),
+        ],
+    )
+    def test_cache_map_traverse_cmap(
```

Comment:
I would recommend adding a couple more cases.  The "root" is special (you've got that covered); the immediate children of the root are reached on the first iteration (you have a "directory" but not a "file"); deeper children require iterating (you have the symlink case which covers this, but it would be nice to have cases for "directory" and "file").

At the very least, it would be good to have at least one case for a "file".  (It's true that some of this is covered by `test_cache_map_get_info_cmap()`, but the functionality here is really in the "`traverse_cmap()` unit" and not in the "`get_info()` unit", so there should be appropriate test cases here.  (The test cases there should be focused on driving behavior which is specific to `get_info()`, and not so much issues around the cache map itself.)